src/main/java/com/ericsson/ei/handlers/UpStreamEventsHandler.java

```diff
@@ -69,9 +69,16 @@ public void runHistoryExtractionRulesOnAllUpstreamEvents(String aggregatedObject
 
         // Use aggregatedObjectId as eventId since they are the same for start
         // events.
-        final ResponseEntity responseEntity = eventRepositoryQueryService
+    	
+    	long start = System.currentTimeMillis();
+    	final ResponseEntity responseEntity = eventRepositoryQueryService
                 .getEventStreamDataById(aggregatedObjectId, SearchOption.UP_STREAM, -1, -1, true);
-        if (responseEntity == null) {
+        
+        long end = System.currentTimeMillis();
+        long milliSec = end - start;
+        
+        LOGGER.info("#### Response time for upstream query for id: {}: {} ", aggregatedObjectId, milliSec);    	
```

Comment:
I guess this is logged often, change to logger debug.

Reply:
WIP